generator/src/main/java/com/linkedin/pegasus/generator/FileFormatDataSchemaParser.java

```diff
@@ -92,8 +92,10 @@ public FileFormatDataSchemaParser(String resolverPath, DataSchemaResolver schema
               while (entries.hasMoreElements())
               {
                 final JarEntry entry = entries.nextElement();
-                if (!entry.isDirectory() && entry.getName().endsWith(_schemaParserFactory.getLanguageExtension()) &&
-                    (entry.getName().startsWith(_schemaResolver.getSchemasDirectoryName().getName()) || isExtensionEntry(entry)))
+                if (!entry.isDirectory() &&
+                    entry.getName().endsWith(_schemaParserFactory.getLanguageExtension()) &&
+                    (entry.getName().startsWith(SchemaDirectoryName.PEGASUS.getName()) ||
```

Comment:
I'd like to make a comment on the original condition logic itself. It took me a while to understand what this if-statement is checking for because it's unnecessarily convoluted. The original logic basically said:

if `entryName.startsWith(schemasDirectoryName)` or `isExtensionEntry()`
...where `isExtensionEntry()` checks that the schemas directory name is `extensions` and that the entry name starts with `extensions/`

Couldn't this just be reduced to

if `entryName.startsWith(schemasDirectoryName + '/')`? Since if the schema directory name is `extensions`, then it would be checking that it starts with `extensions/` anyway

Ignoring the fact that the name `isExtensionEntry()` is misleading and inaccurate (since it also checks the resolver's schemas directory name value, not just the entry), it doesn't even seem like this method is needed.

Since you're changing this logic, though, my suggestion is that you simplify this if-statement to make it more comprehensible.

Reply:
@evanw555 , I modified the logic, please have a look